### PR TITLE
Ports: Add port for xz

### PR DESCRIPTION
**Pull request by gunnarbeutner**
The `libpthread` changes are necessary to make the `xz` port properly detect `pthread` support. The two functions are only used in the configure script and nowhere else.

This port depends on #6343, i.e. won't build properly without it because `-pthread` isn't supported.

### CODE REVIEW

File: Userland/Libraries/LibPthread/pthread.cpp [hunk 1]
```diff
@@ -143,6 +143,19 @@ void pthread_exit(void* value_ptr)
     exit_thread(value_ptr);
 }
 
+void pthread_cleanup_push(void (*routine)(void *), void *arg)
+{
+    (void)routine;
```
Comment (ADKaster) [inline Userland/Libraries/LibPthread/pthread.cpp:148]: The Serenity style for this would be to leave the parameters unnamed in the function signature, I think. Also, should this go in a separate commit, like `LibPthread: add stubs for pthread_cleanup_push/pop`?

Comment (gunnarbeutner) [reply Userland/Libraries/LibPthread/pthread.cpp:148]: Aye, you're right. Will get this fixed in a bit. :+1:

Comment (gunnarbeutner) [reply Userland/Libraries/LibPthread/pthread.cpp]: There we go. :)
